rosbag2_py/src/rosbag2_py/_transport.cpp

```diff
@@ -40,11 +40,11 @@ namespace
 
 rclcpp::QoS qos_from_handle(const py::handle source)
 {
-  auto py_capsule = PyObject_CallMethod(source.ptr(), "get_c_qos_profile", "");
-  const auto rmw_qos_profile = reinterpret_cast<rmw_qos_profile_t *>(
-    PyCapsule_GetPointer(py_capsule, "rmw_qos_profile_t"));
-  const auto qos_init = rclcpp::QoSInitialization::from_rmw(*rmw_qos_profile);
-  return rclcpp::QoS{qos_init, *rmw_qos_profile};
+  PyObject * obj = PyObject_CallMethod(source.ptr(), "get_c_qos_profile", "");
```

Comment:
Can `PyObject *` be `auto` here?

Reply:
it could - but written that way I found it less readable, because it was unclear what types we were working with, I'd prefer to leave it explicitly declared